codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -140,26 +137,19 @@ public QueryStringDecoder(URI uri, Charset charset) {
      * specified charset.
      */
     public QueryStringDecoder(URI uri, Charset charset, int maxParams) {
-        if (uri == null) {
-            throw new NullPointerException("getUri");
-        }
-        if (charset == null) {
-            throw new NullPointerException("charset");
-        }
-        if (maxParams <= 0) {
-            throw new IllegalArgumentException(
-                    "maxParams: " + maxParams + " (expected: a positive integer)");
-        }
+        checkNotNull(uri, "uri");
+        checkNotNull(charset, "charset");
+        checkPositive(maxParams, "maxParams");
 
         String rawPath = uri.getRawPath();
         if (rawPath != null) {
             hasPath = true;
         } else {
-            rawPath = "";
+            rawPath = EMPTY_STRING;
             hasPath = false;
         }
         // Also take care of cut of things like "http://localhost"
-        this.uri = rawPath + (uri.getRawQuery() == null? "" : '?' + uri.getRawQuery());
+        this.uri = rawPath + (uri.getRawQuery() == null? EMPTY_STRING : '?' + uri.getRawQuery());
```

Comment:
Isn't there a useless intermediate concat when hitting the second branch?

Would this be better?
`uri.getRawQuery() == null? rawPath : rawPath + "?" + uri.getRawQuery()`